tldw_Server_API/app/core/Evaluations/recipes/dataset_snapshot.py

```
@@ -0,0 +1,29 @@
+import hashlib
+import json
+from dataclasses import dataclass
+from typing import Any
+
+
+def build_dataset_snapshot_ref(dataset_id: str, dataset_version: str | int) -> str:
+    """Build a stable dataset snapshot reference."""
+    return f"{dataset_id}@v{dataset_version}"
+
+
+def build_dataset_content_hash(payload: Any) -> str:
```

Ruff (0.15.7)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `payload`

(ANN401)